traffic_ops/testing/api/v4/origins_test.go

```diff
@@ -374,3 +473,411 @@ func DeleteTestOrigins(t *testing.T) {
+	}
+	if len(originByProfile.Response) < 1 {
+		t.Fatalf("Expected atleast one Origin for GET Origin by Profile, but found %d", len(originByProfile.Response))
+	}
+
+	//Get Origins by Primary
+	isPrimary := *originByName.Response[0].IsPrimary
+	opts.QueryParameters.Del("profileId")
+	opts.QueryParameters.Set("isPrimary", strconv.FormatBool(isPrimary))
+	originByPrimary, _, err := TOSession.GetOrigins(opts)
+	if err != nil {
+		t.Errorf("cannot get Origin by Primary ID: %v - alerts: %+v", err, originByPrimary.Alerts)
+	}
+	if len(originByPrimary.Response) < 1 {
+		t.Fatalf("Expected atleast one Origin for GET Origin by Primary, but found %d", len(originByPrimary.Response))
+	}
+
+	//Get Origins by Tenant
+	tenant := *originByName.Response[0].TenantID
+	opts.QueryParameters.Del("isPrimary")
+	opts.QueryParameters.Set("tenant", strconv.Itoa(tenant))
+	originByTenant, _, err := TOSession.GetOrigins(opts)
+	if err != nil {
+		t.Errorf("cannot get Origin by Tenant ID: %v - alerts: %+v", err, originByTenant.Alerts)
+	}
+	if len(originByTenant.Response) < 1 {
+		t.Fatalf("Expected atleast one Origin for GET Origin by Tenant, but found %d", len(originByTenant.Response))
+	}
+}
+
+func GetTestOriginsByInvalidParams(t *testing.T) {
+
+	opts := client.NewRequestOptions()
+	opts.QueryParameters.Set("deliveryservice", "12345")
+	originByDs, _, _ := TOSession.GetOrigins(opts)
+	if len(originByDs.Response) > 0 {
+		t.Fatalf("Expected empty response for GET Origin by invalid Delivery Service, but found %d", len(originByDs.Response))
+	}
+
+	//Get Origins by Cachegroup
+	opts.QueryParameters.Del("deliveryservice")
+	opts.QueryParameters.Set("cachegroup", "12345")
+	originByCg, _, _ := TOSession.GetOrigins(opts)
+	if len(originByCg.Response) > 0 {
+		t.Fatalf("Expected empty response for GET Origin by invalid Cachegroups, but found %d", len(originByCg.Response))
+	}
+
+	//Get Origins by Coordinate
+	opts.QueryParameters.Del("cachegroup")
+	opts.QueryParameters.Set("coordinate", "12345")
+	originByCoordinate, _, _ := TOSession.GetOrigins(opts)
+	if len(originByCoordinate.Response) > 0 {
+		t.Fatalf("Expected empty response for GET Origin by invalid Coordinate, but found %d", len(originByCoordinate.Response))
+	}
+
+	//Get Origins by Profile
+	opts.QueryParameters.Del("coordinate")
+	opts.QueryParameters.Set("profileId", "12345")
+	originByProfile, _, _ := TOSession.GetOrigins(opts)
+	if len(originByProfile.Response) > 0 {
+		t.Fatalf("Expected empty response for GET Origin by invalid Profile, but found %d", len(originByProfile.Response))
+	}
+
+	//Get Origins by Tenant
+	opts.QueryParameters.Del("profileId")
+	opts.QueryParameters.Set("tenant", "12345")
+	originByTenant, _, _ := TOSession.GetOrigins(opts)
+	if len(originByTenant.Response) > 0 {
+		t.Fatalf("Expected empty response for GET Origin by invalid Tenant, but found %d", len(originByTenant.Response))
+	}
+}
+
+func CreateTestOriginInvalidData(t *testing.T) {
```

Comment:
In this function, shouldn't one test for err first, before checking for status code?, L645, L658, L671, L684, L697, L710, L723

Reply:
Do you want me to change the order. Either fail, expected would be the same.

Reply:
No, I'm ok with order. Normally, I would check for error first before check for any response or reqInf.